lib/sync/CoreIdCache.js

```diff
@@ -0,0 +1,66 @@
+/*
+* @module CoreIdCache
+*/
+export class CoreIdCache {
+  #sqlite
+  #tableName = 'coreIdCache'
+  #types = `'auth', 'data', 'blob'`
+  /**
+    * Create a CoreIdCache instance. This class handles a table to track cores belonging to different namespaces/stores (Authstore, Datastore, Blobstore)
+    * @param {import('../sqlite.js').Sqlite} sqlite an instance of better-sqlite3
+  */
+  constructor(sqlite){
+    this.#sqlite = sqlite
```

Comment:
I'm not sure it makes sense to use our Sqlite class here? It does extra work parsing links and forks fields, which don't exist here, and creates an abstraction that makes it harder to parse what is happening. It also does not allow us to pre-prepare the statement to improve efficiency of gets.